snap/hooks/install

```diff
@@ -1,28 +1,43 @@
-#!/bin/bash
+#!/usr/bin/env bash
+# Installation hook for charmed-mongodb snap
 
 set -eux
 
+export CONF="${SNAP_DATA}"/etc/mongod
+export PBM_CONF="${SNAP_DATA}"/etc/pbm
+export DATA="${SNAP_COMMON}"/var/lib/mongodb
+export LOGS="${SNAP_COMMON}"/var/log/mongodb
+export MONGO_CONFIG_FILE="${CONF}"/mongod.conf
+
 # Create the necessary parent directories 
-mkdir -p "${SNAP_DATA}/etc/pbm/"
-mkdir -p "${SNAP_DATA}/etc/mongod/"
-mkdir -p "${SNAP_COMMON}/var/lib/mongodb/"
-mkdir -p "${SNAP_COMMON}/var/log/mongodb/"
+mkdir -p $DATA
+mkdir -p $LOGS
+mkdir -p $PBM_CONF
+mkdir -p $CONF
+
+if [ stat -c '%u' "$DATA" == 0 -o ! -d "$DATA" ]; then
+    chmod -R 770 "${SNAP_COMMON}"
+    chmod 750 "${DATA}"
+fi
+
+if [ stat -c '%u' "$LOGS" == 0 -o ! -d "$LOGS" ]; then
+    chmod -R 770 "${SNAP_COMMON}"
+    chmod g+s "$LOGS"/*
+fi
+
+chown -R 584788:root "${SNAP_COMMON}"/*
 
 # Copy over the mongod.conf and create needed directories/files
-MONGO_CONFIG_FILE="${SNAP_DATA}/etc/mongod/mongod.conf"
 echo "configuration file does not exist."
 echo "copying default config to ${MONGO_CONFIG_FILE}"
 cp -r ${SNAP}/etc/mongod.conf ${MONGO_CONFIG_FILE}
 
 
 # mongod.conf default values are not consistent with the snap directory system.
 sed -i "s/fork: true/fork: false/g" $MONGO_CONFIG_FILE
-sed -i "s:/var/log/mongodb:$SNAP_COMMON/var/log/mongodb:g" $MONGO_CONFIG_FILE
-sed -i "s:/var/lib/mongodb:$SNAP_COMMON/var/lib/mongodb:g" $MONGO_CONFIG_FILE
+sed -i "s:/var/log/mongodb:$LOGS:g" $MONGO_CONFIG_FILE
+sed -i "s:/var/lib/mongodb:$DATA:g" $MONGO_CONFIG_FILE
 sed -i "s:/var/run:/tmp:g" $MONGO_CONFIG_FILE
 
 # Change ownership of snap directories to allow snap_daemon to read/write
-chmod g+s "${SNAP_COMMON}/var/log/"*
 chown -R 584788:root "${SNAP_DATA}"/*
-chown -R 584788:root "${SNAP_COMMON}"/*
-chgrp root "${SNAP_COMMON}/var/log/"*
```

Comment:
I assume this is part of the problem and thas why it is getting removed?

Reply:
No, this was just duplicated instructions, and it's still run on line 28 :)
The main issue was that in case the files had the wrong owner, it would fail to go on the install script.
This is fixed by the `stat` check to check the owner and chmod if needed.
I just moved a few things around to clean the snap.

Reply:
Where is `chgrp root `? I don't see it .

But also I am not sure if it is needed. I dont know if root should have any ownership

Reply:
Oh the `chgrp root` was duplicated instruction, duplicated by `chown -R 584788:root "${SNAP_COMMON}"/*` which gives the `root` group to the log directory so it can be safely removed.

Reply:
thank you